### ISSUE TITLE
Graph for rolling average temperature additionally to absolute per-hour numbers

### ISSUE THREAD

**Issue by Currrupted** [martykan/forecastie#343]
The weather graphs are very helpful and a nice way to illustrate the weather in future. However, I think that it might be a great idea to show a graph of how the temperature might change in future, by displaying the **general** temperature in addition to the raw temperature values.

Right in the "Temperature" section, the temperatures are displayed on a per-hour bases. When these values change regularly and drastically (for example in spring where there are great differences in night vs. day temperatures), it is hard to properly comprehend in which way the temperature is _actually_ changing.

To achieve this, the current diagram for the temperature could include two graphs, one for the absolute values, and one showing broader values which might change over the course of a day or a week,

To me, the best way to implement is to calculate average temperatures and display them similarly to the graph that is already existing.

**Comment by robinpaulson** [comment]
This sounds like you are requesting a moving average; would that be correct? Those are indeed useful. We could easily add it to the temperature graph without causing too much clutter, but I am wary what will happen when we move all the graphs to one chart (see #297 ).

**Comment by Currrupted** [comment]
> 
> 
> This sounds like you are requesting a moving average (https://en.wikipedia.org/wiki/Moving_average), would that be correct? Those are indeed useful. We could easily add it to the temperature graph without causing too much clutter, but I am wary what will happen when we move all the graphs to one chart (see #297 ).

Yeah, exactly! That is what I meant. I absolutely agree that it may be wise to move this to another chart. Unfortunately I do not have the statistical knowledge to implement this easily, but I may check it out in near future and I will see what I can do.

**Comment by robinpaulson** [comment]
The statistics are fairly simple, rather than taking the value at the present time, take the average of the last n values. n is configurable, the larger it is, the more "smooth" the graph will be.

Having said it's useful, I'm going to go back on that and say I'm not sure what the utility would be. Given that we only use a 5 day forecast (more than that is of dubious accuracy), I'm not sure what trends you're going to pick up that wouldn't be trivial to see anyway. If I look at the temperature graph, I can easily see if it will generally trend up over the next few days.

I'm going to close this and mark it "won't fix", but do add more comments here if you think there is a compelling case and I may consider reopening.

**Comment by Currrupted** [comment]
> 
> 
> The statistics are fairly simple, rather than taking the value at the present time, take the average of the last n values. n is configurable, the larger it is, the more "smooth" the graph will be.
> 
> Having said it's useful, I'm going to go back on that and say I'm not sure what the utility would be. Given that we only use a 5 day forecast (more than that is of dubious accuracy), I'm not sure what trends you're going to pick up that wouldn't be trivial to see anyway. If I look at the temperature graph, I can easily see if it will generally trend up over the next few days.
> 
> I'm going to close this and mark it "won't fix", but do add more comments here if you think there is a compelling case and I may consider reopening.

I absolutely get your point and I do not think that this feature should be prioritized over other, more urgent, issues.

However, I do think that average and projected temperature changes could significantly improve the comprehensibility of the weather forecasts. 

Additionally, I could think of customized recommendations, based on the projections for the next hours and/or days.  For example, if the rolling average of air temperature steeply declines throughout the next three days, users could be warned with messages such as "Brace for cold weather" or "Time to use that winter coat!".

![grafik](https://user-images.githubusercontent.com/13292623/69996676-d1d32a00-1552-11ea-839f-9af61255802f.png) ([Source](https://csform.com/wp-content/uploads/2018/10/Weather-App-Pro-Ionic.jpg))

I would be willing to contribute code for this feature and would be finished in roughly two months.

**Comment by robinpaulson** [comment]
We are making some changes in the near future, particularly to the graphs. See #397 

There is also an open issue for warnings #192 

That said, I think there is utility in your suggestions, so I'll reopen this and we can discuss more.
